### PR TITLE
busybox: Save shell history into /tmp

### PR DESCRIPTION
**Pull request by hnyman**
Enable the busybox feature to save shell command history. (.apk size increase 0.5 kB)

To prevent flash wear,
* save history only at the exit from a shell session, and
* set /tmp as the default location for the history file. The history is kept on ramdisk until a reboot, when the history is then lost.

If the user wants to save history onto flash, he can change the location definition in /etc/profile.d/busybox-history-file.sh

run-tested on MT-6000, DL-WRX36, RT3200

### CODE REVIEW

File: package/utils/busybox/files/busybox-history-file.sh [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export HISTFILE=/tmp/.busybox_ash_history
```
Comment (systemcrash) [inline package/utils/busybox/files/busybox-history-file.sh:1]: Does the resulting file need any special permissions treatment?

Comment (hnyman) [reply package/utils/busybox/files/busybox-history-file.sh:1]: Well, busybox creates the history file with tight permissions, owner only:

```
root@router6000:~# ls -l /tmp/.busybox_ash_history
-rw-------    1 root     root            67 Dec  5 23:17 /tmp/.busybox_ash_history
```

Comment (systemcrash) [reply package/utils/busybox/files/busybox-history-file.sh:1]: OK. That would be my only concern then. Looks good.

Would it make sense to save it to flash upon reboot in `/root`? Although what you have here is 'good enough'.

Comment (hnyman) [reply package/utils/busybox/files/busybox-history-file.sh:1]: My proposal is intentionally ramdisk-only. 
Like I wrote in commit message, if somebody wants to save history over reboots, they can easily change the location without recompiling. /etc/profile.d/ is also saved over reboot, so any change you make there, gets carried over.

**Comment by Ansuel** [comment]
I know this is minimal but wonder if we should make this configurable and just an option in system uci similar to how we do for log size?

What is the max size of the history? Remember stuff in /tmp = less ram

**Comment by hnyman** [comment]
The default is just 255 history items in the history buffer, so likely not that many kB.
(Max is 9999 items)

It can be controlled with a shell variable, so in theory also that variable could be added to the profile item.
But I doubt that most users exceed that 255 items before the next sysupgrade or reboot.

(The basic feature needs to be compiled into busybox and is then controlled by shell variables, so UCI controlling is not that easy. Feel free to tweak, if you want to implement that...)

**Review by ynezz — APPROVED** [review]
LGTM, you've just removed one item from my TODO list, thanks!

**Comment by hnyman** [comment]
Yeah, I have carried this snippet for years in my own builds, but I am now trying to upstream my small modifications...

**Pull request merged**